guava/src/com/google/common/primitives/UnsignedLongs.java

```diff
@@ -65,6 +65,7 @@ private UnsignedLongs() {}
    * longs, that is, {@code a <= b} as unsigned longs if and only if {@code flip(a) <= flip(b)} as
    * signed longs.
    */
+  @SuppressWarnings("signedness:return")
   private static @PolySigned long flip(@PolySigned long a) {
```

Comment:
I think this should be:
```java
  private static @Signed long flip(@Unsigned long a) {
```
In addition to more accurately reflecting the method's specification, that should also reduce the number of signedness warnings.

Reply:
It actually increases the number of warnings as the authors use 'flip' to go both from unsigned to signed and from signed back to unsigned.  See min, max and sort methods.   Since flip is private to the class,  I think I will leave it as PolySigned.